#3004: Email member on portfolio permissions updates [BACKUP]

dave-kennedy-ecs opened the pull request:
## Ticket

<!-- PR title format: `#issue_number: Descriptive name ideally matching ticket name - [sandbox]`-->
Resolves #3004 

## Changes

<!-- What was added, updated, or removed in this PR. -->
- Send email to member (not invited member) of a portfolio, when their permissions are changed through the registrar.
- In the case that the email fails to send for any reason, a warning message is presented to the user, but the action still completes.

<!--
    Please add/remove/edit any of the template below to fit the needs
    of this specific PR.
--->

## Context for reviewers

<!--Background context, more in-depth details of the implementation, and anything else you'd like to call out or ask reviewers.  -->

## Setup

<!--  Add any steps or code to run in this section to help others run your code.
    
    Example 1:
    ```sh
    echo "Code goes here"
    ``` 
    
    Example 2: If the PR was to add a new link with a redirect, this section could simply be:
    -go to /path/to/start/page
    -click the blue link in the <insert location>
    -notice user is redirected to <proper end location>
-->

## Code Review Verification Steps

### As the original developer, I have

#### Satisfied acceptance criteria and met development standards

- [x] Met the acceptance criteria, or will meet them in a subsequent PR
- [x] Created/modified automated tests
- [x] Update documentation in READMEs and/or onboarding guide

#### Ensured code standards are met (Original Developer)
<!-- Mark "- N/A" and check at the end of each check that is not applicable to your PR -->
- [x] If any updated dependencies on Pipfile, also update dependencies in requirements.txt.
- [x] Interactions with external systems are wrapped in try/except
- [x] Error handling exists for unusual or missing values

#### Validated user-facing changes (if applicable)

- [x] Tag @dotgov-designers in this PR's Reviewers for design review. If code is not user-facing, delete design reviewer checklist
- [x] Verify new pages have been added to .pa11yci file so that they will be tested with our automated accessibility testing
- [x] Checked keyboard navigability
- [x] Tested general usability, landmarks, page header structure, and links with a screen reader (such as Voiceover or ANDI)

### As a code reviewer, I have

#### Reviewed, tested, and left feedback about the changes

- [ ] Pulled this branch locally and tested it
- [ ] Verified code meets all checks above. Address any checks that are not satisfied
- [ ] Reviewed this code and left comments. Indicate if comments must be addressed before code is merged
- [ ] Checked that all code is adequately covered by tests
- [ ] Verify migrations are valid and do not conflict with existing migrations

#### Validated user-facing changes as a developer
**Note:** Multiple code reviewers can share the checklists above, a second reviewer should not make a duplicate checklist. All checks should be checked before approving, even those labeled N/A. 

- [ ] New pages have been added to .pa11yci file so that they will be tested with our automated accessibility testing
- [ ] Checked keyboard navigability
- [ ] Meets all designs and user flows provided by design/product
- [ ] Tested general usability, landmarks, page header structure, and links with a screen reader (such as Voiceover or ANDI)
- [ ] (Rarely needed) Tested as both an analyst and applicant user

### As a designer reviewer, I have

#### Verified that the changes match the design intention

- [ ] Checked that the design translated visually
- [ ] Checked behavior. Comment any found issues or broken flows.
- [ ] Checked different states (empty, one, some, error)
- [ ] Checked for landmarks, page heading structure, and links

#### Validated user-facing changes as a designer

- [ ] Checked keyboard navigability
- [ ] Tested general usability, landmarks, page header structure, and links with a screen reader (such as Voiceover or ANDI)
- [ ] Tested with multiple browsers (check off which ones were used)
  - [ ] Chrome
  - [ ] Microsoft Edge
  - [ ] FireFox
  - [ ] Safari
- [ ] (Rarely needed) Tested as both an analyst and applicant user

### References
- [Code review best practices](../docs/dev-practices/code_review.md)

## Screenshots

<!-- If this PR makes visible interface changes, an image of the finished interface can help reviewers
and casual observers understand the context of the changes.
A before image is optional and can be included at the submitter's discretion.

Consider using an animated image to show an entire workflow.
You may want to use [GIPHY Capture](https://giphy.com/apps/giphycapture) for this! 📸

_Please frame images to show useful context but also highlight the affected regions._
--->

Katherine-Osos commented:
@dave-kennedy-ecs I added "gov designers" to the reviewer list. The Design Review for member management is already underway, so this email will need to be reviewed by design as part of this PR. Sorry for any confusion!

dave-kennedy-ecs commented on src/registrar/templates/emails/portfolio_update.txt:
```diff
@@ -0,0 +1,35 @@
+
+ORGANIZATION: {{ portfolio.organization_name }}
+UPDATED BY: {{ requestor_email }}
+UPDATED ON: {{ date }}
```
@zandercymatics thanks, I fixed the error with the date. It was not being passed to the email.

dave-kennedy-ecs commented:
@Katherine-MN @allly-b - I have dev approval on this one. Can I merge or do I need to wait?

Katherine-MN commented:
@dave-kennedy-ecs This looks great! One small edit related to something we changed during the m5 design review. For "Domains" permissions, we update the "Viewer, all" permission to simply read "Viewer." So the two options that can display in the email are:
- Viewer, limited
- Viewer

Maybe that change will automatically be applied to this email when those changes from the design review are merged. If so, then I'll approve this PR as is.

dave-kennedy-ecs commented:
> @dave-kennedy-ecs This looks great! One small edit related to something we changed during the m5 design review. For "Domains" permissions, we update the "Viewer, all" permission to simply read "Viewer." So the two options that can display in the email are:
> 
> * Viewer, limited
> * Viewer
> 
> Maybe that change will automatically be applied to this email when those changes from the design review are merged. If so, then I'll approve this PR as is.

@Katherine-MN I made the change from Viewer, all to Viewer in this PR. The changes made in other branches won't automatically apply, because I refactored in this ticket so that the same permissions display values would appear in multiple places, but be managed in one singular place.
